test(e2e): messaging.e2e fails on main — send without mention now rejected (400)

## Symptom

On current `main`, `packages/e2e/src/tests/messaging.e2e.test.ts` fails:

```
M2 messaging — user-scoped HTTP chat send + replyTo
  › sends a text message and gets a 201 with a fresh message id
  › replyTo threads the second message onto the first via inReplyTo
AssertionError: expected 400 to be 201
```

Reproduce: `E2E_WITH_CLIENT=1 pnpm --filter @first-tree/e2e vitest run --config vitest.config.ts src/tests/messaging.e2e.test.ts`

## Root cause

The test posts to `POST /api/v1/chats/:chatId/messages` with `{ format, content }` and no recipient routing:

```js
body: JSON.stringify({ format: "text", content: "hello from e2e" }),
```

The group-chat policy now requires an explicit recipient — the server responds 400:

> Sending a message requires an explicit recipient. Pass `metadata.mentions: [agentId]` (or `receiverNames: [name]`), or set `purpose: "agent-final-text"` for silent history-only sends.

The test predates that rule and was never updated, so it's red on `main` independent of any feature branch.

## Fix

Update the two sends in `messaging.e2e.test.ts` to declare routing — `metadata.mentions: [testAgentId]` (matching how the new TUI e2e fixture sends, see `packages/e2e/src/framework/runtime-tui-fixture.ts` in #748). Re-run to confirm 201 + the `inReplyTo` threading assertions still hold.

## Origin

Pre-existing failure surfaced while validating PR5 TUI e2e (#748). Not touched by that PR (it sends with mentions); filed separately so the mention-rule reconciliation for the legacy SDK messaging test is tracked on its own.

--- a/packages/e2e/src/tests/messaging.e2e.test.ts
+++ b/packages/e2e/src/tests/messaging.e2e.test.ts
@@ -76,7 +76,15 @@ describe("M2 messaging — user-scoped HTTP chat send + replyTo", () => {
     const res = await fetch(`${handle.serverBaseUrl}/api/v1/chats/${chatId}/messages`, {
       method: "POST",
       headers: { "Content-Type": "application/json", Authorization: `Bearer ${creds.accessToken}` },
-      body: JSON.stringify({ format: "text", content: "hello from e2e" }),
+      // Group-chat policy requires an explicit recipient — declare routing via
+      // `metadata.mentions: [agentUuid]`, the same channel the client runtime
+      // uses (see services/message.ts + runtime-tui-fixture.ts). Without it the
+      // send is rejected 400.
+      body: JSON.stringify({
+        format: "text",
+        content: "hello from e2e",
+        metadata: { mentions: [testAgentId] },
+      }),
     });
     expect(res.status).toBe(201);
     const body = (await res.json()) as { id: string; chatId: string; senderId: string };
@@ -93,7 +101,7 @@ describe("M2 messaging — user-scoped HTTP chat send + replyTo", () => {
     const first = await fetch(`${handle.serverBaseUrl}/api/v1/chats/${chatId}/messages`, {
       method: "POST",
       headers: { "Content-Type": "application/json", Authorization: `Bearer ${creds.accessToken}` },
-      body: JSON.stringify({ format: "text", content: "parent" }),
+      body: JSON.stringify({ format: "text", content: "parent", metadata: { mentions: [testAgentId] } }),
     });
     expect(first.status).toBe(201);
     const firstBody = (await first.json()) as { id: string };
@@ -101,7 +109,12 @@ describe("M2 messaging — user-scoped HTTP chat send + replyTo", () => {
     const reply = await fetch(`${handle.serverBaseUrl}/api/v1/chats/${chatId}/messages`, {
       method: "POST",
       headers: { "Content-Type": "application/json", Authorization: `Bearer ${creds.accessToken}` },
-      body: JSON.stringify({ format: "text", content: "child", inReplyTo: firstBody.id }),
+      body: JSON.stringify({
+        format: "text",
+        content: "child",
+        inReplyTo: firstBody.id,
+        metadata: { mentions: [testAgentId] },
+      }),
     });
     expect(reply.status).toBe(201);
     const replyBody = (await reply.json()) as { id: string };

